core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -140,6 +140,7 @@
   protected final Map<String, SqlNode> ordinalMap = new HashMap<>();
 
   protected final Map<CorrelationId, Context> correlTableMap = new HashMap<>();
+  protected boolean isTableNameColumnNameIdentical = false;
```

Comment:
Please avoid class level variable. pass it as parameter if required.

Reply:
As discussed over the call with Yogesh Sir and Soma, resolving this as a new object of SqlImplementor is created per reltree